resources/monitor.rb

```diff
@@ -1,16 +1,56 @@
 # Configure a service via its yaml file
 
-actions :add, :remove
-default_action :add if defined?(default_action)
+default_action :add
 
-attribute :name, :kind_of => String, :name_attribute => true
-attribute :cookbook, :kind_of => String, :default => 'datadog'
+property :name, String, name_attribute: true
+property :cookbook, String, default: 'datadog'
 
 # checks have 3 sections: init_config, instances, logs
 # we mimic these here, no validation is performed until the template
 # is evaluated.
-attribute :init_config, :kind_of => Hash, :required => false, :default => {}
-attribute :instances, :kind_of => Array, :required => false, :default => []
-attribute :logs, :kind_of => Array, :required => false, :default => []
-attribute :version, :kind_of => Integer, :required => false, :default => nil
-attribute :use_integration_template, :kind_of => [TrueClass, FalseClass], :required => false, :default => false
+property :init_config, [Hash, nil], required: false, default: {}
+property :instances, Array, required: false, default: []
+property :version, [Integer, nil], required: false, default: nil
+property :use_integration_template, [TrueClass, FalseClass], required: false, default: false
+property :logs, [Array, nil], required: false, default: []
+
+action :add do
+  log "Adding monitoring for #{new_resource.name}" do
+    level :debug
+  end
+
+  template ::File.join(node['datadog']['config_dir'], 'conf.d', "#{new_resource.name}.yaml") do
+    if node['platform_family'] == 'windows'
+      owner 'Administrators'
+      rights :full_control, 'Administrators'
+      inherits false
```

Comment:
@remeh this will have to be updated in light of your changes here: https://github.com/DataDog/chef-datadog/pull/588